src/common/config_opts.h

```diff
@@ -899,6 +899,7 @@ OPTION(osd_min_pg_log_entries, OPT_U32, 3000)  // number of entries to keep in t
 OPTION(osd_max_pg_log_entries, OPT_U32, 10000) // max entries, say when degraded, before we trim
 OPTION(osd_force_recovery_pg_log_entries_factor, OPT_FLOAT, 1.3) // max entries factor before force recovery
 OPTION(osd_pg_log_trim_min, OPT_U32, 100)
+OPTION(osd_pg_log_dup_track_versions, OPT_U32, 1000) // how many versions back to track operations that could be duplicated in the pg log
```

Comment:
We should think through how this interacts with the full pg log entries a little more. With the defaults here, the 1000 dup ops are just pointless work since the pg log entries are always longer. I see a few choices:
1) Turn the pg log entries *way* down when the PG is clean; only keep them around for log-based recovery when the PG is degraded. Increase dup op size accordingly.
2) run without dup ops by default, and expect people turn them up when turning down the pg log sizes.
3) Switch the config options around so that we have a "minimum tail of dup op detection" and auto-tune it so that we keep around as many dup ops as needed for that (which might be none, if the log is long enough, or if we want to "remember" the last 2000 ops and are only maintaining the last 1000 log entries, we switch them when evicting or something).